src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/HandlerMethodDetails.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// This is consumed as 'generated' code in a source package and therefore requires an explicit nullable enable
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reflection;
+using System.Threading;
+
+namespace Microsoft.CommonLanguageServerProtocol.Framework;
+
+internal sealed record HandlerMethodDetails(
```

Comment:
The ordering of "handler" and "method" was intentional due to a couple of complications:

1. There is a one-to-many relationship between a "handler type" and its "handler methods". A handler type can implement many handler interfaces, each of which will have a "Handles" method, the details of which are capture by this type.\
2. There are two different "methods" referred to by CLaSP. First, the "Method" in `IMethodHandler` refers to an LSP "method" as is defined by the LSP protocol. This type refers to the actual C# method that will be called via reflection.

That said, if this is unclear or I used it inconsistently, I'm perfectly happy to clean it up to avoid future CLaSP breaks. Given the complications above, do you have other suggestions?

Reply:
I can't come up with anything better, so works for me.  But could consider adding a doc comment with a small blurb detailing what it is and how it maps to a specific handler instance.

Reply:
Looking at the code again, I think I originally convinced myself of the wrong thing. This really does represent the details of a specific `IMethodHandler`. I'll go ahead and rename it to `MethodHandlerDetails` and fix up any other relevant names.